lib/IDE/SwiftSourceDocInfo.cpp

```diff
@@ -883,6 +884,18 @@ struct RangeResolver::Implementation {
       return llvm::any_of(StartMatches, IsCase) &&
           llvm::any_of(EndMatches, IsCase);
     }
+
+    bool isMultiDecl() {
+      if (StartMatches.empty() || EndMatches.empty())
+        return false;
+
+      // Multi-statement should have the same nominal type as a common parent
+      if (auto ParentDecl = Parent.dyn_cast<Decl *>())
+        if (isa<NominalTypeDecl>(ParentDecl))
+          return true;
```

Comment:
`return isa<NominalTypeDecl>(ParentDecl);` is good enough here.

Reply:
What happens if we select multiple top-level decls? It seems `MultiDecl` actually mean `MultiTypeMemberDecl`. If this is true, could we rename it?